File: src/main/java/io/piveau/importing/MainVerticle.java
package io.piveau.importing;

import io.piveau.importing.rdf.ImportingRdfVerticle;
import io.piveau.pipe.connector.PipeConnector;
import io.vertx.core.AbstractVerticle;
import io.vertx.core.DeploymentOptions;
import io.vertx.core.Launcher;
import io.vertx.core.Promise;

import java.util.Arrays;

public class MainVerticle extends AbstractVerticle {

    @Override
    public void start(Promise<Void> startPromise) {
        vertx.deployVerticle(ImportingRdfVerticle.class, new DeploymentOptions().setWorker(true).setConfig(config()), result -> {
            if (result.succeeded()) {
                PipeConnector.create(vertx, cr -> {
                    if (cr.succeeded()) {
                        cr.result().consumer(ImportingRdfVerticle.ADDRESS);
                        startPromise.complete();
                    } else {
                        startPromise.fail(cr.cause());
                    }
                });
            } else {
                startPromise.fail(result.cause());
            }
        });
    }

    public static void main(String[] args) {
        String[] params = Arrays.copyOf(args, args.length + 1);
        params[params.length - 1] = MainVerticle.class.getName();
        Launcher.executeCommand("run", params);
    }

}

File: src/main/kotlin/io/piveau/importing/MainVerticle.kt
package io.piveau.importing

import io.piveau.importing.rdf.ImportingRdfVerticle
import io.piveau.pipe.connector.PipeConnector
import io.vertx.core.DeploymentOptions
import io.vertx.core.Launcher
import io.vertx.kotlin.coroutines.CoroutineVerticle
import io.vertx.kotlin.coroutines.await
import java.nio.file.FileAlreadyExistsException

class MainVerticle : CoroutineVerticle() {

    override suspend fun start() {

        if (!vertx.fileSystem().exists("tmp").await()) {
            vertx.fileSystem().mkdir("tmp").await()
        } else if (!vertx.fileSystem().props("tmp").await().isDirectory) {
            throw FileAlreadyExistsException("tmp")
        }

        vertx.deployVerticle(ImportingRdfVerticle::class.java, DeploymentOptions().setWorker(true)).await()

        PipeConnector.create(vertx, DeploymentOptions()).await().publishTo(ImportingRdfVerticle.ADDRESS)
    }

}

fun main(args: Array<String>) {
    Launcher.executeCommand("run", *(args.plus(MainVerticle::class.java.name)))
}
